dart/lib/src/utils/http_header_utils.dart

```diff
@@ -0,0 +1,16 @@
+import 'package:meta/meta.dart';
+
+/// Helper to extract header data
+@internal
+class HttpHeaderUtils {
+  /// Get `Content-Length` header
+  static int? getContentLength(Map<String, List<String>> headers) {
```

Comment:
Can we just read directly from the `response` instead?
Eg `final length = response.contentLength`.